h2o-core/src/main/java/water/Scope.java

```diff
@@ -33,17 +34,12 @@
    *  enter call except for the listed Keys.
    *  @return Returns the list of kept keys. */
   static public Key[] exit(Key... keep) {
-    List<Key> keylist = new ArrayList<>();
-    if( keep != null )
-      for( Key k : keep ) if (k != null) keylist.add(k);
-    Object[] arrkeep = keylist.toArray();
-    Arrays.sort(arrkeep);
+    Set<Key> mustKeep = new HashSet<>(Arrays.asList(keep));
     Stack<HashSet<Key>> keys = _scope.get()._keys;
-    if (keys.size() > 0) {
+    if (!keys.empty()) {
       Futures fs = new Futures();
       for (Key key : keys.pop()) {
-        int found = Arrays.binarySearch(arrkeep, key);
-        if ((arrkeep.length == 0 || found < 0) && key != null) Keyed.remove(key, fs);
+        if (!mustKeep.contains(key)) Keyed.remove(key, fs);
       }
       fs.blockForPending();
     }
```

Comment:
This line is problematic too... Because if `keys` is empty, then we are not supposed to call `exit()` at all -- it's like an unbalanced curly brace... http://xkcd.com/859/